include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -226,6 +140,25 @@ struct aws_s3_meta_request {
          * initially be 1 for part 1, and after that part is received, it will be 2, then 3, etc.. */
         uint32_t next_streaming_part;
 
+        /* Number of parts scheduled for delivery. */
+        uint32_t num_parts_delivery_sent;
+
+        /* Total number of parts that have been attempted to be delivered. (Will equal the sum of succeeded and
+         * failed.)*/
+        uint32_t num_parts_delivery_completed;
+
+        /* Number of parts that have been successfully delivered to the caller. */
+        uint32_t num_parts_delivery_succeeded;
+
+        /* Number of parts that have failed while trying to be delivered to the caller. */
+        uint32_t num_parts_delivery_failed;
+
+        /* The end finish result of the meta request. */
+        struct aws_s3_meta_request_result finish_result;
+
+        /* True if the finish result has been set. */
+        uint32_t finish_result_set : 1;
```

Comment:
Is this necessary? Wouldn't the result be the way to detect this?

Reply:
aws_s3_meta_request_result is visible to the user, so wanted to keep whether it was populated or not outside of the structure.